xlators/storage/posix/src/posix-helpers.c

```diff
@@ -2276,7 +2276,7 @@ posix_spawn_health_check_thread(xlator_t *xl)
     {
         /* cancel the running thread  */
         if (priv->health_check_active == _gf_true) {
-            pthread_cancel(priv->health_check);
+            (void)pthread_cancel(priv->health_check);
```

Comment:
I think it's better to test the return value as it indicates a real system error.
From `pthread_cancel `man page
```
the return status of pthread_cancel() merely informs the caller whether the cancellation request was successfully queued.
 ```
Not sure if we should skip the creation of the new thread but at least logged an ERROR msg

Reply:
It's true that on a theoretical level the return value should be checked, though if such fundamental operations fails, something is probably very wrong with the system and recovery is not likely (similarly, we do check for allocation failures, but can we really recover from OOM? no sure at all).
Regardless, logging an error seems fair, I'll modify.